### PR TITLE
d8vk: new verb

### PR DESCRIPTION
**Pull request by sonic2kk**
Adds d8vk verb as requested in #2004.

Tested with Touhou 6 via Lutris, MangoHud showed the renderer changed from WINED3D to DXVK. The selection of D3D8 games I can test with is somewhat limited it seems so further testing is of course welcome.

This PR touches `helper_dxvk`, I was specifically concerned with the changes to the `tar` command which now uses `--one-top-level`, so I tested installing regular `dxvk` and it seemed to work fine. This extra flag is required because DXVK archives have a root folder in the archive, but D8VK does not.

One outstanding issue is with the URL in `helper_dxvk` that points to the release notes; this URL is incorrect for D8VK. This is because DXVK release tags are just the version, e.g. `v2.1`, but D8VK is `d8vk-vX.X`. The `_W_package_version` needs to be set differently and I wasn't sure of the best approach to take with this.

First time contribution and first time even looking at Winetricks code so any and all feedback is very welcome :-)

Thanks!

### CODE REVIEW

File: src/winetricks [hunk 1]
```diff
@@ -6912,14 +6912,14 @@ helper_dxvk()
...
-        w_try tar -zxf "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}"
+        w_try tar -zxf "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}" --one-top-level
     fi
 
     for _W_dll in ${_W_dll_overrides}; do
         w_try mv "${W_TMP}/${_W_package_dir}/x32/${_W_dll}.dll" "${W_SYSTEM32_DLLS}/"
         [ "${_W_dll}" = "d3d9" ] && _W_d3d9_support="1"
     done
-    if test "${W_ARCH}" = "win64"; then
+    if test "${W_ARCH}" = "win64" && -d "${W_TMP}/${_W_package_dir}/x64"; then
```
Comment (sonic2kk) [inline src/winetricks:6922]: Direct3D 8 is only for 32bit applications, so the archive only has a `x32` directory, hence this check.

**Comment by sonic2kk** [comment]
I think the checks should pass now

**Comment by sonic2kk** [comment]
That was embarrassing, sorry... Guess I changed something since the last time I ran shellcheck or something.

Passes on my machine now, gave the same warning that the CI did so I guess it's fixed :slightly_smiling_face:

File: src/winetricks [hunk 2]
```diff
@@ -6912,14 +6912,14 @@ helper_dxvk()
         w_try_unzip "${W_TMP}" "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}"
     else
         w_try_cd "${W_TMP}"
-        w_try tar -zxf "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}"
+        w_try tar -zxf "${W_CACHE}/${W_PACKAGE}/${_W_package_archive}" --one-top-level
```
Comment (austin987) [inline src/winetricks:6915]: `--one-top-level` is a GNU extension; this will break on BSD tar.

Comment (sonic2kk) [reply src/winetricks:6915]: Thanks! After some light research it looks like `-C` might work for BSD tar but I haven't tested just yet.

Comment (sonic2kk) [reply src/winetricks]: Updated to use `-C "${_W_package_dir}"`, Winetricks still fetches and installs d8vk correctly (tested with no cached archive and with a cached archive), only tested Touhou 6 at the moment and it still works too.

**Comment by sonic2kk** [comment]
Since merging master the diff seems like it's gotten pretty big? It mentions removing `helper_dxvk_nvapi` which I saw when resolving the conflicts, but I just chose to accept the incoming changes which evidently removed it. This function is also not present on master it looks like from a Ctrl+F.

Not sure what went wrong there :frowning:

**Comment by austin987** [comment]
There was a merge (dxvk_nvapi) that conflicted, so it needs a manual merge.

I did some initial fixes (﻿https://github.com/austin987/winetricks/commit/01c9188bc8683e483be40f6138e68cd65d464bc9), but didn't tested it with a real application.

Note that the release notes are still broken (should be fixed). Also, this will always point to the latest dxvk release; you probably want to add at least one real release so users have a stable option they can use.

**Comment by sonic2kk** [comment]
Not sure why this is only working for some games and crashing others. Maybe the supported games list only refers to specific versions of games. Touhou 6 works for me, but Syberia II does not. The D3D9 DLL gets overwritten for all the games it looks like, but Touhou 6 works still. I tested different Wine-GE versions and Lutris Wine 7.2 and d8vk consistently installed properly. 

I noticed the install script will move the DLLs into the Proton lib folders for Steam games as well as into the games prefix, but as far as I can tell this is not something Winetricks does at all, so I don't think it'a step required here.

Maybe that limits the usefulness of including d8vk in Winetricks. I am not sure what would be going wrong with the code, as it mostly appears to mirror how DXVK is installed, and D8VK works for at least one game.

**Pull request closed without merging**